lib/setup/setupSetup.js

```diff
@@ -54,18 +56,96 @@ function Setup(options) {
         }
     }
 
-    function setupReady(callback) {
+    async function informAboutPlugins(systemConfig, callback) {
+        let ioPackage;
+        let ioConfig;
+
+        const configFile = tools.getConfigFileName();
+        try {
+            ioPackage = JSON.parse(fs.readFileSync(path.join(__dirname, '../../io-package.json'), 'utf8'));
+        } catch(err) {
+            console.error('Can not read js-controller io-package.json. Ignore plugins defined there.');
+            ioPackage = null;
+        }
+        try {
+            ioConfig = JSON.parse(fs.readFileSync(configFile, 'utf8'));
+        } catch(err) {
+            console.error('Can not read js-controller config file. Ignore plugins defined there.');
+            ioConfig = null;
+        }
+
+        const plugins = {};
+        if (ioPackage && ioPackage.common && ioPackage.common.plugins) {
+            Object.keys(ioPackage.common.plugins).forEach(plugin => {
+                if (ioPackage.common.plugins[plugin].enabled !== false) {
+                    plugins[plugin] = ioPackage.common.plugins[plugin];
+                }
+            });
+        }
+        if (ioConfig && ioConfig.plugins) {
+            Object.keys(ioConfig.plugins).forEach(plugin => {
+                if (!plugins[plugin] && ioConfig.plugins[plugin].enabled !== false) {
+                    plugins[plugin] = ioConfig.plugins[plugin];
+                }
+            });
+        }
+
+        let systemLang = 'en';
+        let systemDiag = 'extended';
+        if (systemConfig && systemConfig.common) {
+            systemDiag = systemConfig.common.diag || 'extended';
+            systemLang = systemConfig.common.language || 'en';
+        }
+
+        const pluginNames = Object.keys(plugins);
+        for (let i = 0; i < pluginNames.length; i++) {
+            const plugin = pluginNames[i];
+            if (!pluginInfos.PLUGIN_INFOS[plugin]) {
```

Comment:
You are accessing `pluginInfos.PLUGIN_INFOS[plugin]` ca. 5 times - I think that warrants a variable.